### DISCUSSION TITLE
Is it worth pursuing further?

### DISCUSSION THREAD

**Discussion by pygy** [jivanpal/apfs-tools#77] [Q&A] [answered]
Hi there,

I lost a bit of source code in a freak `rm -r` accident (really, a `fs-extra` `fs.remove()` mishap in NodeJS).

I rebooted the Mac in Asahi Linux (I had installed it previously) after a couple of minutes (long enough for the newly freed space to be overwritten by my confused browsing, even if unlikely).

I've dabbled a bit with `drat` which gives a lot of information I don't know how to decode...

So before digging further, I wrote a small C program to look for known strings in the partition data. After a couple of minutes, the script dumped quite a few chunks of data, but all of them were false positives (i.e. the needles were found but it was not my code).

Given these circumstances, do I stand a chance of recovering the lost files?

Edit: I'm not holding my breath, obviously...

**Comment by jivanpal** [comment]
Is the APFS container encrypted (is/was FileVault enabled in macOS)? If so, Drat can't help you right now.

Is the drive in question an SSD, and if so, is/was [TRIM](https://en.wikipedia.org/wiki/Trim_(computing)) enabled on the system? If so, it's possible/likely that the blocks containing your file data were trimmed away by the SSD controller some time after the files were deleted.

I'm not in a position to provide direct support with data recovery nowadays, but I am hoping to resume development of Drat within the next few weeks, hopefully focusing on a guide on how to use it effectively.

**Comment by pygy** [comment, marked as answer]
Thanks for your reply. This is my main machine, and the Asahi Linux install was crippled and barely functional, so I decided to accept the loss and move on.
